build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/BwcVersions.java

```diff
@@ -252,6 +252,19 @@ private List<Version> getReleased() {
             .toList();
     }
 
+    public List<Version> getReadOnlyIndexCompatible() {
+        // Lucene can read indices in version N-2
```

Comment:
I wonder if we should assert that N-2 == 7? We definitely want to utiliize this for the V9 upgrade, but we are not fully committed to necessarily having N-2 support in V10 (not saying we won't either).

Reply:
I added an explicit check with a comment in the test suite. I think it achieves the same goal, while avoiding it to be in the build logic.